onnx/test/cpp/parser_test.cc

```diff
@@ -557,5 +557,31 @@ TEST(ParserTest, TypesModelTest2) {
   CheckModel(code);
 }
 
+TEST(ParserTest, ExternalDataTest) {
+  const char* code = R"ONNX(
+agraph (float y = {1.0}, float[N] z) => (w) <
+    float[3, 2] m1 = ["location": "weight_1.bin", "offset": "17"],
```

Comment:
Just thinking aloud: wonder if we should allow `location` without the quotations. Advantage is that it is a bit simpler and more natural. Disadvantage is that, technically, this is a string field and could be a non-identifier (in future extensions), so using quotes is more general. Anyway, I am fine with this for now.

Reply:
I have exactly the same wondering, too. The syntax without quotations is more natural. But after I read the [Textual Syntax](docs/Syntax.md) document, I notice that the whole grammar uses no keyword. It only use symbols. I am not sure it is intentional or not. But I am afraid to break this rule.

Reply:
The syntax document does not mention them. But they are used in particular contexts, like when specifying `ir_version` or `opset_import` for a model. See [here](https://github.com/onnx/onnx/blob/60b0a34037f2980c519793b1bbc3a6ed79b0dfa3/onnx/defs/parser.h#L122). However, they are not keywords, in the sense users can still use those words as names of tensors if they wish. They are interpreted specially only in specific contexts.

But this is fine for now. I think we can update this later if we want.

Reply:
Wow. Thank you. I did not notice that. If I know this earlier I probably use keywords. We can change this later :D